ios/RNJudo.m

```diff
@@ -65,34 +93,101 @@ - (dispatch_queue_t)methodQueue
-    paymentAuthorizationViewController.delegate = self;
+RCT_REMAP_METHOD(showPaymentMethods,
+                 options:(NSDictionary *)options
+                 showPaymentMethodsResolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self initWithOptions:options];
 
-    self.applePayResolve = resolve;
-    self.applePayReject = reject;
+    PaymentMethods paymentMethod = [RCTConvert NSUInteger:options[@"paymentMethods"]];
+    JudoKit *judoKit = [self judoKit];
+    ApplePayConfiguration *applePayConfiguration = [self appleConfigWith: paymentMethod and: options];
+    JPAmount *judoAmount = [[JPAmount alloc] initWithAmount:self.amount currency:self.currency];
 
-    UIViewController *rootViewController = UIApplication.sharedApplication.keyWindow.rootViewController;
-    [rootViewController presentViewController:paymentAuthorizationViewController animated:YES completion:nil];
+    [judoKit invokePayment:self.judoId
+                    amount:judoAmount
+         consumerReference:self.consumerReference
+            paymentMethods:paymentMethod
+   applePayConfiguratation:applePayConfiguration
+               cardDetails:nil
+                completion:[self paymentCompletion:judoKit reject:reject resolve:resolve]];
+}
+
+- (JudoKit *) judoKit
+{
+    JudoKit *judoKit = [[JudoKit alloc] initWithToken:self.token secret:self.secret];
+    judoKit.apiSession.sandboxed = self.isSandbox;
+    return judoKit;
+}
+
+- (JPReference *) generateReferenceWith:(NSString *) consumerReference
+                       paymentReference:(NSString *) paymentReference
+                               metaData:(NSDictionary *) metaData
+{
+    JPReference *judoReference = [[JPReference alloc] initWithConsumerReference:consumerReference
+                                                               paymentReference:paymentReference];
+    [judoReference setMetaData:metaData];
+    return judoReference;
+}
+
+- (ApplePayConfiguration *) appleConfigWith:(PaymentMethods) paymentMethod
+                                        and:(NSDictionary *)options
+{
+    ApplePayConfiguration *applePayConfiguration = nil;
+    if (paymentMethod == PaymentMethodApplePay || paymentMethod == PaymentMethodsAll) {
+        PaymentShippingType paymentShippingType = [RCTConvert NSInteger:options[@"paymentShippingType"]];
+        TransactionType transactionType = [RCTConvert NSInteger:options[@"transactionType"]];
+        NSString *countryCode = [RCTConvert NSString:options[@"countryCode"]];
+        NSString *merchantIdentifier = [RCTConvert NSString:options[@"merchantId"]];
+
+        NSMutableArray<PaymentSummaryItem *> * paymentSummaryItems = [NSMutableArray new];
+        for (NSDictionary *dict in options[@"paymentSummaryItems"]) {
+          [paymentSummaryItems addObject: [[PaymentSummaryItem alloc] initWithLabel:dict[@"label"]
```

Comment:
Still not great indentation. The tab size should be 4, not 2 but whatevs, man.

Reply:
👍